nni/algorithms/compression/pytorch/pruning/dependency_aware_pruner.py

```diff
@@ -0,0 +1,159 @@
+# Copyright (c) Microsoft Corporation.
+# Licensed under the MIT license.
+
+import logging
+import torch
+from schema import And, Optional, SchemaError
+from nni.common.graph_utils import TorchModuleGraph
+from nni.compression.pytorch.utils.shape_dependency import ChannelDependency, GroupDependency
+from nni.compression.pytorch.utils.config_validation import CompressorSchema
+from nni.compression.pytorch.compressor import Pruner
+from .constants import MASKER_DICT
+
+__all__ = ['DependencyAwarePruner']
+
+logger = logging.getLogger(__name__)
+logger.setLevel(logging.INFO)
+
+
+class DependencyAwarePruner(Pruner):
```

Comment:
@zheng-ningxin is this pruner written by you? where is the previous version?

Reply:
Yes, it's at https://github.com/microsoft/nni/blob/9444e27508ab9d6e4184436af2661cd4efd387bd/nni/algorithms/compression/pytorch/pruning/structured_pruning.py#L70